src/main/scala/org/broadinstitute/hail/methods/LogisticRegression.scala

```diff
@@ -0,0 +1,91 @@
+package org.broadinstitute.hail.methods
+
+import breeze.linalg._
+import org.apache.spark.rdd.RDD
+import org.broadinstitute.hail.Utils._
+import org.broadinstitute.hail.annotations.Annotation
+import org.broadinstitute.hail.stats.LogisticRegressionModel
+import org.broadinstitute.hail.variant._
+import org.broadinstitute.hail.expr._
+
+object LogisticRegression {
+  def name = "LogisticRegression"
+
+  def buildX(gs: Iterable[Genotype], C: DenseMatrix[Double]): Option[DenseMatrix[Double]] = {
+    require(gs.size == C.rows)
+
+    val (nCalled, gtSum) = gs.flatMap(_.gt).foldRight((0,0))((gt, acc) => (acc._1 + 1, acc._2 + gt))
+
+    // allHomRef || allHomVar || allNoCall || allHet
+    if (gtSum == 0 || gtSum == 2 * nCalled || nCalled == 0 || (gtSum == nCalled && gs.flatMap(_.gt).forall(_ == 1)))
+      None
+    else {
+      val gtMean = gtSum.toDouble / nCalled
+      val gtArray = gs.map(_.gt.map(_.toDouble).getOrElse(gtMean)).toArray
+      Some(DenseMatrix.horzcat(new DenseMatrix(gtArray.length, 1, gtArray), C))
+    }
+  }
+
+  def apply(vds: VariantDataset, y: DenseVector[Double], cov: Option[DenseMatrix[Double]], stat: String): LogisticRegression = {
+    require(cov.forall(_.rows == y.length))
+    require(y.forall(yi => yi == 0 || yi == 1))
+    require{ val sumY = sum(y); sumY > 0 && sumY < y.size }
```

Comment:
Is this how IntelliJ is indenting?  I'd expect a space before the opening curly brace.